vespa_analysis/vespa_eval.py

```diff
@@ -0,0 +1,159 @@
+def pair_map (gen_dict,ref_dict,mmseq_out,vespa_out,threads):
+    ref_id = ''
+    ref_seq = ''
+    gen_id = ''
+    gen_seq = ''
+
+    with open ('vespa_pair_cmd.list','w') as cmd:
+        for i in range(len(gen_dict)):
+            ref_id = list(ref_dict.keys())[i]
+            ref_seq = ref_dict[ref_id]
+            gen_id = list(gen_dict.keys())[i]
+            gen_seq = gen_dict[gen_id]
+            cmd.write('python vespa_pair.py --mmseq_out '+mmseq_out+' --vespa_out '+vespa_out+'/'+str(i)+'.csv --ref_id '+ref_id+' --ref_seq '+ref_seq+' --gen_id '+gen_id+' --gen_seq '+gen_seq+'\n')
+
+    os.system('parallel -j '+str(threads)+' < vespa_pair_cmd.list')
+
+## Function for MSA Mapping VESPA scores and Conservation Classes ##
+def msa_map (ref_fasta,gen_fasta,threads):
+    ref_gen_msa = generate_msa(ref_fasta,gen_fasta,threads)
+    msa_dict = fasta2dict(ref_gen_msa)
+    cons_fasta = cwd+'/vespa_run_directory/conspred_class.fasta'
+    cons_dict = fasta2dict(cons_fasta)
+    mapped_scores = cwd+'/vespa_scores'
+
+    with open ('vespa_msa_cmd.list','w') as cmd:
+        for i in range(len(gen_dict)):
+            ref_id = list(ref_dict.keys())[i]
+            gen_id = list(gen_dict.keys())[i]
+            ref_aln = msa_dict[ref_id]
+            gen_aln = msa_dict[gen_id]
+            cons_class = cons_dict[ref_id]
+            vespa_map_out = cwd+'/vespa_scores/'+gen_id+'_vespa_scores.csv'
+            cmd.write('python vespa_msa.py --vespa_map_out '+vespa_map_out+' --ref_id '+ref_id+' --ref_aln '+ref_aln.replace('-','_')+' --cons_class '+cons_class+' --gen_id '+gen_id+' --gen_aln "'+gen_aln.replace('-','_')+'"\n')
+    os.system('parallel -j '+str(threads)+' < vespa_msa_cmd.list')   
+    os.system('cat vespa_msa/cons_stats/*.csv > vespa_msa/'+gen_name+'_cons_scores.csv')
+    os.system('rm -r vespa_msa/cons_stats/')
+    os.system('cat vespa_msa/sav_stats/*.csv > vespa_msa/'+gen_name+'_sav_scores.csv')
+    os.system('rm -r vespa_msa/sav_stats/')
+
+parser = argparse.ArgumentParser(description='Calculates SAV scores using VESPA')
+parser.add_argument('-r', type=str, action='store', help='FULL Path Fasta file containing the Reference sequences',required=True)
+parser.add_argument('-g', type=str, action='store', help='FULL Path Fasta file containing the Generated sequences',required=True)
+parser.add_argument('--cache', type=str, action='store', help='FULL Path of caching folder to store Prott5 weights',required=True)
+parser.add_argument('--threads', type=str, action='store', help='Number of threads to be used in parallelization',required=True)
+args = parser.parse_args()
+
+## Parsing Input ##
+ref_fasta = args.r
+ref_name = ref_fasta.split('/')[-1]
+ref_name = ref_name.split('.')[0]
+gen_fasta = args.g 
+gen_name = gen_fasta.split('/')[-1]
+gen_name = gen_name.split('.')[0]
+cache_path = args.cache
+threads = int(args.threads)
+cwd = os.getcwd()
+
+ref_dict = fasta2dict(ref_fasta)
+gen_dict = fasta2dict(gen_fasta)
+
+dummy_thread = 0
```

Comment:
we need to group this block into a function with relevant name to be clear for anyone to know what is happening here. What do you think?